Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -62,3 +65,41 @@ public class OpenFeatureAPI {
         self.hooks.removeAll()
     }
 }
+
+// MARK: Provider Events
+
+extension OpenFeatureAPI {
+
+    public func addHandler(observer: Any, selector: Selector, event: ProviderEvent) {
+        providerNotificationCentre.addObserver(
+            observer,
+            selector: selector,
+            name: event.notification,
+            object: nil
+        )
+    }
+
+    public func removeHandler(observer: Any, event: ProviderEvent) {
+        providerNotificationCentre.removeObserver(observer, name: event.notification, object: nil)
+    }
+
+    public func emitEvent(
+        _ event: ProviderEvent,
+        provider: FeatureProvider,
+        error: Error? = nil,
```

Comment:
q: what does it mean to emit an error? what's the use case? why we can't just emit events? basically just passing `providerEvent`

Reply:
I think the code is simply emitting a `ProviderEvent`. Error is optional field to add to notification in case of failure. So in case of error `ProviderEvent` will be emitted with dictionary containing additional information about an error.

Reply:
huh, to be transparent. I jumped into theses conversations to align my understanding of it, so feedback from @Calibretto is welcome 🙂

Reply:
What @alina-v1 mentioned above is accurate - it is just emitting an event, but you can also pass an `Error` if you want, simply to add more information (it's not specifically required in the specification, but it does mention that you can send additional information, such as a message if you want to)